### PR TITLE
picosha2: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **picosha2/***

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — DISMISSED** [review]
LGTM

File: recipes/picosha2/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+from conan import ConanFile, conan_version
```
Comment (prince-chrismc) [inline recipes/picosha2/all/conanfile.py:1]:
```suggestion
from conan import ConanFile
```

File: recipes/picosha2/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        basic_layout(self, src_folder="src")
+
+    def package_id(self):
+        self.info.clear()
+
+    @property
+    def _info(self):
+        return self if Version(conan_version).major < 2 else self.info
+
+    def validate(self):
```
Comment (prince-chrismc) [inline recipes/picosha2/all/conanfile.py:34]:
```suggestion
    def validate(self):
```

File: recipes/picosha2/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        if self._info.settings.compiler.get_safe("cppstd"):
```
Comment (prince-chrismc) [inline recipes/picosha2/all/conanfile.py:35]:
```suggestion
        if self.info.settings.compiler.get_safe("cppstd"):
```

**Review by prince-chrismc — CHANGES_REQUESTED** [review]
This limitation will be fixed in an upcomming release of the client, we like to limit itis usage 

[(there was a PR for this)](https://github.com/conan-io/conan-center-index/pull/13754) puts it back to 1.x syntax and the beta 5 will be improved

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**